html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php

```diff
@@ -74,6 +145,69 @@ public function unsubscribe(AccountInterface $user) {
     throw new AccessDeniedHttpException();
   }
 
+  /**
+   * Unsubscribe a user (one click).
+   *
+   * @param string $opaque
+   *   Opaque data from the unsubscribe link.
+   *
+   * @return array
+   *   Render array for the page.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
+   *   Bad request if the link is invalid.
+   */
+  public function unsubscribeOneClick(string $opaque): array {
+    $data = $this->mailer->decryptOneClickUnsubscribeLink($opaque);
+    if ($data === FALSE) {
+      throw new BadRequestHttpException();
+    }
+
+    [$uid, $sid] = $data;
+
+    $subscriptions = reliefweb_subscriptions_subscriptions();
+    if (!isset($subscriptions[$sid])) {
+      throw new BadRequestHttpException();
+    }
+
+    $user = $this->getEntityTypeManager()->getStorage('user')->load($uid);
+    if (is_null($user)) {
+      throw new BadRequestHttpException();
+    }
+
+    // Remove the subscription.
+    $this->database->delete('reliefweb_subscriptions_subscriptions')
+      ->condition('sid', $sid)
+      ->condition('uid', $user->id())
+      ->execute();
+
+    $subscription_page = '/user/' . $user->id() . '/notifications';
+
+    if ($user->id() === $this->getCurrentUser()->id()) {
+      $message = $this->t('To add new subscriptions or manage your list, please go to your <a href=":subscription_page">subscriptions</a> page.', [
+        ':subscription_page' => $subscription_page,
+      ]);
+    }
+    else {
+      $login_link = Url::fromRoute('user.login', [], [
+        'query' => ['destination' => $subscription_page],
+      ]);
+
+      $message = $this->t('To add new subscriptions or manage your list, please <a href=":login_link">log in</a>.', [
+        ':login_link' => $login_link->toString(),
+      ]);
+    }
+
+    return [
+      '#type' => 'inline_template',
+      '#template' => '<p><strong>You have successfully unsubscribed.</strong></p><p>You will not longer receive <em>{{ subscription_label }}</em> notifications.</p><p>{{ message }}</p>',
```

Comment:
Typo: "You will no longer receive" (without the 't')

Reply:
thanks.